src/app/properties/property.service.ts

```diff
@@ -1,20 +1,44 @@
-import { Injectable } from '@angular/core';
-import {Observable, of} from "rxjs";
-import Property from "../shared/models/property.model";
-import Location from "../shared/models/location.model";
+import {Injectable} from '@angular/core';
+import {Observable} from "rxjs";
+import {Property} from "../shared/models/property.model";
 import {HttpClient} from "@angular/common/http";
 import {environment} from "../../env/environment";
 import {ApiPaths} from "../shared/api/api-paths.enum";
+import PropertiesSearchDto from "./properties-search.dto";
+import {Router, UrlSerializer} from "@angular/router";
 
 @Injectable({
   providedIn: 'root'
 })
 export class PropertyService {
 
-  constructor(private http: HttpClient) { }
+  constructor(
+    private http: HttpClient,
+    private router: Router,
+    private serializer: UrlSerializer,
+  ) { }
 
-  getProperties(): Observable<Property[]> {
-    return this.http.get<Property[]>(`${environment.apiHost}/${ApiPaths.Properties}`);
+  getProperties(searchDto?: PropertiesSearchDto): Observable<Property[]> {
+    let url = `${environment.apiHost}/${ApiPaths.Properties}`;
+
+    if (searchDto) {
+      const adaptedSearchDto = {
+        ...searchDto,
+        startDate: searchDto?.startDate?.toISOString(),
+        endDate: searchDto?.endDate?.toISOString(),
+      }
```

Comment:
This will have to be removed soon as I have added an interceptor that does this.
Since the interceptor is not available in this PR, no action is needed on this right now.